controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -252,14 +255,83 @@ func (r *OCSInitializationReconciler) ensureRookCephOperatorConfigExists(initial
 			Namespace: initialData.Namespace,
 		},
 	}
-	err := r.Client.Create(r.ctx, rookCephOperatorConfig)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		r.Log.Error(err, fmt.Sprintf("Failed to create %s configmap", util.RookCephOperatorConfigName))
+
+	csiPluginDefaults := defaults.DaemonPlacements["csi-plugin"]
+	csiProvisionerDefaults := defaults.DaemonPlacements["csi-provisioner"]
+
+	pluginTolerations := r.getCsiPluginTolerations()
+	provisionerTolerations := r.getCsiProvisionerTolerations()
+
+	opResult, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rookCephOperatorConfig, func() error {
+
+		if rookCephOperatorConfig.Data == nil {
+			rookCephOperatorConfig.Data = make(map[string]string)
+		}
+
+		if pluginTolerations != nil {
+			tolerations := append(pluginTolerations, csiPluginDefaults.Tolerations...)
+			tolerationsYAML, err := yaml.Marshal(tolerations)
+			if err != nil {
+				return err
+			}
+			rookCephOperatorConfig.Data["CSI_PLUGIN_TOLERATIONS"] = string(tolerationsYAML)
+			initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified = true
+		} else if pluginTolerations == nil && initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified {
+			delete(rookCephOperatorConfig.Data, "CSI_PLUGIN_TOLERATIONS")
+			initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified = false
+		}
+
+		if provisionerTolerations != nil {
+			tolerations := append(provisionerTolerations, csiProvisionerDefaults.Tolerations...)
+			tolerationsYAML, err := yaml.Marshal(tolerations)
+			if err != nil {
+				return err
+			}
+			rookCephOperatorConfig.Data["CSI_PROVISIONER_TOLERATIONS"] = string(tolerationsYAML)
+			initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified = true
+		} else if provisionerTolerations == nil && initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified {
+			delete(rookCephOperatorConfig.Data, "CSI_PROVISIONER_TOLERATIONS")
+			initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified = false
+		}
```

Comment:
can we do something like below to avoid application?
```suggestion
updateTolerationsConfig := func(tolerations,defaults []Toleration, key string, modifiedFlag *bool) error {
        if tolerations == nil && !*modifiedFlag {
            return nil
        }
        updatedTolerations := append(tolerations, defaults...)
        tolerationsYAML, err := yaml.Marshal(updatedTolerations)
        if err != nil {
            return err
        }

        if tolerations == nil {
            delete(rookCephOperatorConfig.Data, key)
        } else {
            rookCephOperatorConfig.Data[key] = string(tolerationsYAML)
        }

        *modifiedFlag = tolerations != nil
        return nil
    }
 if err := updateTolerationsConfig(pluginTolerations,csiPluginDefaults.Tolerations, "CSI_PLUGIN_TOLERATIONS", &initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified); err != nil {
        return err
    }

    if err := updateTolerationsConfig(provisionerTolerations,csiProvisionerDefaults.Tolerations, "CSI_PROVISIONER_TOLERATIONS", &initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified); err != nil {
        return err
    }
```

Reply:
Done, changed a bit to minimize if else